### ISSUE TITLE
Change it to GPL Part 2

### ISSUE THREAD

**Issue by goob256** [aseprite/aseprite#3613]
I contributed the initial Mac port of Allegro Sprite Editor back when you didn't have a Mac. My code is licensed GPL and you never consulted anyone about changing the license. So as far as anyone can be concerned, Aseprite is GPL anyway.

**Comment by goob256** [comment]
https://imgur.com/gallery/yqOqe94

**Comment by DevChimera** [comment]
As of now, Aseprite uses the [AsepriteEULA](https://github.com/aseprite/aseprite/blob/main/EULA.txt). When you contribute to a project in Github, the list of contributors is automatically listed in the project's [GitHub](https://github.com/aseprite/aseprite/graphs/contributors). As you can imagine, manually listing all contributors is not a possible thing, as Aseprite is open source and gets continuously contributed. 

Now, the side note, if you contributed during the GPL time of the code, you accepted that the code licensing is irrevocable as it's stated in [GPL](https://www.gnu.org/licenses/gpl-3.0.en.html) ("...and are irrevocable provided the stated conditions are met.")

TL;DR, Aseprite has a general "thanks to contributors" list on GitHub, and if you contributed code with GPL, it means you allow its use irrevocably. 

_Important note, I'm not a lawyer and I do not represent legal representation for neither of the parties involved._

**Comment by goob256** [comment]
The EULA is just that. For End Users. I had never run this application until David asked me to port it to Mac. We knew each other a little bit and he knew I had a Mac. So I did.

**Comment by goob256** [comment]
https://imgur.com/gallery/rzKZJFU

**Comment by DevChimera** [comment]
Sorry it seems you didn't quite understand, here's the quote directly from source:

" Source code and official releases/binaries are distributed under our End-User License Agreement for Aseprite (EULA). Please check that there are modules/libraries in the source code that are distributed under the MIT license (e.g. laf, clip, undo, observable, ui, etc.). "

And again, by contributing, you accept the terms of the program.

**Comment by goob256** [comment]
You'll conveniently unsubscribe before I ask, when did this license take effect and where is the commit hash? Thanks.

**Comment by dacap** [comment]
Hi @goob256, before the license changed we switched the backend from Allegro to our own code+Skia library on Windows and macOS (and then on Linux, removing Allegro dependency), several changes were completely removed.

About those commits, a8fcc8b0af02668d9241d5d05b54ef7f41fd2d15, [is not used anymore](https://github.com/aseprite/aseprite/commit/7750982af6159d0bcb01336da915d93c88318886#diff-a6402e0c899a19f3f5c68ea8dccf642320173dc7c414c0c9643f06a2b16efdd0R236) (and sorry to say this, but I would highly doubt about its contribution importance to request a license change):

```diff
commit a8fcc8b0af02668d9241d5d05b54ef7f41fd2d15
Author: David Capello <davidcapello@gmail.com>
Date:   Fri Aug 13 11:35:43 2010 -0300

    Patch contributed by Trent Gamblin to compile ASE with Mac OS X.

diff --git a/src/file/png_format.cpp b/src/file/png_format.cpp
index 8a6aa12f2..f2e83d875 100644
--- a/src/file/png_format.cpp
+++ b/src/file/png_format.cpp
@@ -209,7 +209,7 @@ static bool load_PNG(FileOp *fop)
   for (pass = 0; pass < number_passes; pass++) {
     for (y = 0; y < height; y++) {
       /* read the line */
-      png_read_row(png_ptr, row_pointer, png_bytepp_NULL);
+      png_read_row(png_ptr, row_pointer, (png_byte*)NULL);
 
       /* RGB_ALPHA */
       if (info_ptr->color_type == PNG_COLOR_TYPE_RGB_ALPHA) {
```

And 2bc2db1630f248a13363092f31ea574499f26651 which is also completely removed as we are not using Allegro library anymore:

```diff
commit 2bc2db1630f248a13363092f31ea574499f26651
Author: David Capello <davidcapello@gmail.com>
Date:   Mon Aug 17 00:26:05 2009 +0000

    Added support to compile ASE for Mac OS X (thanks to Trent Gamblin).

diff --git a/src/core/dirs.cpp b/src/core/dirs.cpp
index 02923f1b8..19b032d2f 100644
--- a/src/core/dirs.cpp
+++ b/src/core/dirs.cpp
@@ -117,7 +117,7 @@ DIRS *filename_in_datadir(const char *filename)
   DIRS *dirs = dirs_new();
   char buf[1024];
 
-#if defined ALLEGRO_UNIX
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX
 
   /* $HOME/.ase/filename */
   sprintf(buf, ".ase/%s", filename);
@@ -153,7 +153,7 @@ DIRS *filename_in_homedir(const char *filename)
 {
   DIRS *dirs = dirs_new();
 
-#if defined ALLEGRO_UNIX
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX
 
   char *env = getenv("HOME");
   char buf[1024];
@@ -187,7 +187,7 @@ DIRS *cfg_filename_dirs()
 {
   DIRS *dirs = dirs_new();
 
-#if defined ALLEGRO_UNIX
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX
 
   /* $HOME/.asepriterc */
   dirs_cat_dirs(dirs, filename_in_homedir(".asepriterc"));
diff --git a/src/core/file_system.cpp b/src/core/file_system.cpp
index 289a4d9a2..6feed481a 100644
--- a/src/core/file_system.cpp
+++ b/src/core/file_system.cpp
@@ -43,10 +43,10 @@
   #define USE_PIDLS
 #endif
 
-#if defined ALLEGRO_UNIX || defined ALLEGRO_DJGPP || defined ALLEGRO_MINGW32
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX || defined ALLEGRO_DJGPP || defined ALLEGRO_MINGW32
   #include <sys/stat.h>
 #endif
-#if defined ALLEGRO_UNIX || defined ALLEGRO_MINGW32
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX || defined ALLEGRO_MINGW32
   #include <sys/unistd.h>
 #endif
 
diff --git a/src/jinete/jfilesel.cpp b/src/jinete/jfilesel.cpp
index 94734affb..fe832d9cc 100644
--- a/src/jinete/jfilesel.cpp
+++ b/src/jinete/jfilesel.cpp
@@ -36,10 +36,10 @@
 #include <allegro/internal/aintern.h>
 #include <errno.h>
 #include <string.h>
-#if defined ALLEGRO_UNIX || defined ALLEGRO_DJGPP || defined ALLEGRO_MINGW32
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX || defined ALLEGRO_DJGPP || defined ALLEGRO_MINGW32
 #  include <sys/stat.h>
 #endif
-#if defined ALLEGRO_UNIX || defined ALLEGRO_MINGW32
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX || defined ALLEGRO_MINGW32
 #  include <sys/unistd.h>
 #endif
 
diff --git a/src/jinete/jthread.cpp b/src/jinete/jthread.cpp
index b2a536d73..d16bbdd2e 100644
--- a/src/jinete/jthread.cpp
+++ b/src/jinete/jthread.cpp
@@ -39,11 +39,11 @@
 #if defined ALLEGRO_WINDOWS
   #include <winalleg.h>
   #include <process.h>
-#elif defined ALLEGRO_UNIX
+#elif defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX
   #include <pthread.h>
 #endif
 
-#if defined ALLEGRO_UNIX
+#if defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX
 struct pthread_proxy_data {
   void (*proc)(void*);
   void* data;
@@ -65,7 +65,7 @@ JThread jthread_new(void (*proc)(void*), void* data)
 {
 #if defined ALLEGRO_WINDOWS
   return (JThread)_beginthread(proc, 0, data);
-#elif defined ALLEGRO_UNIX
+#elif defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX
   pthread_t thread = 0;
   struct pthread_proxy_data *ptr = jnew(struct pthread_proxy_data, 1);
   ptr->proc = proc;
@@ -84,7 +84,7 @@ void jthread_join(JThread thread)
 {
 #if defined ALLEGRO_WINDOWS
   WaitForSingleObject(thread, INFINITE);
-#elif defined ALLEGRO_UNIX
+#elif defined ALLEGRO_UNIX || defined ALLEGRO_MACOSX
   pthread_join((pthread_t)thread, NULL);
 #else
   #error ASE does not support threads for your platform
diff --git a/src/raster/image_impl.h b/src/raster/image_impl.h
index a8c417b3f..1521c7db5 100644
--- a/src/raster/image_impl.h
+++ b/src/raster/image_impl.h
@@ -503,7 +503,7 @@ void ImageImpl<RgbTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
   switch (depth) {
 
     case 8:
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       if (is_planar_bitmap(bmp)) {
 	for (y=0; y<h; y++) {
 	  bmp_address = (unsigned long)bmp->line[_y];
@@ -536,7 +536,7 @@ void ImageImpl<RgbTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
   
 	  _y++;
 	}
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       }
 #endif
       break;
@@ -635,7 +635,7 @@ void ImageImpl<GrayscaleTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
   switch (depth) {
 
     case 8:
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       if (is_planar_bitmap(bmp)) {
         for (y=0; y<h; y++) {
           bmp_address = (unsigned long)bmp->line[_y];
@@ -663,7 +663,7 @@ void ImageImpl<GrayscaleTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
 
           _y++;
         }
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       }
 #endif
       break;
@@ -766,7 +766,7 @@ void ImageImpl<IndexedTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
   switch (depth) {
 
     case 8:
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       if (is_planar_bitmap (bmp)) {
         for (y=0; y<h; y++) {
           bmp_address = (unsigned long)bmp->line[_y];
@@ -793,7 +793,7 @@ void ImageImpl<IndexedTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
 
           _y++;
         }
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       }
 #endif
       break;
@@ -884,7 +884,7 @@ void ImageImpl<BitmapTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
       color[0] = makecol8(0, 0, 0);
       color[1] = makecol8(255, 255, 255);
 
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       if (is_planar_bitmap(bmp)) {
         for (y=0; y<h; y++) {
 	  addr = line_address(y);
@@ -915,7 +915,7 @@ void ImageImpl<BitmapTraits>::to_allegro(BITMAP *bmp, int _x, int _y) const
 
           _y++;
         }
-#if defined GFX_MODEX && !defined ALLEGRO_UNIX
+#if defined GFX_MODEX && !defined ALLEGRO_UNIX && !defined ALLEGRO_MACOSX
       }
 #endif
       break;
```

**Comment by goob256** [comment]
Those are not the only commits, and those are not all Allegro things, those are just the macros that are provided and thus useful when using Allegro.

**Comment by goob256** [comment]
It's something 95% of professionals wouldn't know how to do, it's my expertise and I wouldn't have provided the free service to "Igura Studio" like I would for a GPL project.

**Comment by goob256** [comment]
Oh well. It's not that much code. I think it only took me a few hours anyway.
